Bottom: 2031b3f08de1c4128cb10032820232efdf65a84c
Top:    86a4cf15e1435780660060112ce6d6c8ad9bfaa9
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 20:31:32 +0100

Workaround crash where contour is passed as NULL to poly_PreContour






---

diff --git a/src/polygon.c b/src/polygon.c
index c697f3e..5c2d1f9 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -202,6 +202,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
